libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -0,0 +1,449 @@
+                                        remainingLength, NULL );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Test connectPacketSize > pBuffer->size. */
+    /* Get MQTT connect packet size and remaining length. */
+    mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                            NULL,
+                                            &remainingLength,
+                                            &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    networkBuffer.pBuffer = mqttBuffer;
+    networkBuffer.size = remainingLength - 1;
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, NULL,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTNoMemory, mqttStatus );
+}
+
+/**
+ * @brief This method calls MQTT_SerializeConnect successfully using different parameters
+ * until we have full coverage on the private method, serializeConnectPacket(...).
+ */
+void test_MQTT_SerializeConnect_happy_paths()
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    size_t remainingLength = 0, packetSize = 0;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTConnectInfo_t connectInfo;
+    MQTTPublishInfo_t willInfo;
+
+    /* Fill structs to pass into methods to be tested. */
+    setupNetworkBuffer( &networkBuffer );
+    setupConnectInfo( &connectInfo );
+    setupWillInfo( &willInfo );
+
+    /* Get MQTT connect packet size and remaining length. */
+    mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                            &willInfo,
+                                            &remainingLength,
+                                            &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Make sure buffer has enough space. */
+    TEST_ASSERT_GREATER_OR_EQUAL( packetSize, networkBuffer.size );
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, &willInfo,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    verifySerializedConnectPacket( &connectInfo, &willInfo,
+                                   remainingLength, &networkBuffer );
+
+    /* Repeat with QoS2. */
+    willInfo.qos = MQTTQoS2;
+    mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                            &willInfo,
+                                            &remainingLength,
+                                            &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Make sure buffer has enough space. */
+    TEST_ASSERT_GREATER_OR_EQUAL( packetSize, networkBuffer.size );
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, &willInfo,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    verifySerializedConnectPacket( &connectInfo, &willInfo,
+                                   remainingLength, &networkBuffer );
```

Comment:
I think a few more cases can be added here to check different branches( Will Info being null, username and  password being null, clean session as false, retain flag in willinfo as false. etc. )